agent/fw_drupal.c

```diff
@@ -588,6 +603,30 @@ NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_before) {
+  (void)wraprec;
+  zval* hook_copy = NULL;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
+
+  hook_copy = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (nr_php_is_zval_non_empty_string(hook_copy)) {
+    nr_stack_push(&NRPRG(drupal_module_invoke_all_hooks),
+                  hook_copy);
+  }
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_after) {
+    (void)wraprec;
+    zval* hook_copy = nr_stack_pop(&NRPRG(drupal_module_invoke_all_hooks));
```

Comment:
Do we just want to peek the top element of the stack, or pop it? The description for this function `nr_stack_pop` states that it's only performing a 'peek' operation.

Reply:
upon further investigation it looks like the description for `nr_stack_pop` is just wrong and was taken directly from `nr_stack_get_top` without being updated

Reply:
Looks like the function description is incorrect. It has the same description as nr_stack_get_top(), which actually does a peek. I'll add a fix to that with my next push.